tools/topology2/sof-cnl-nocodec.conf

```diff
@@ -0,0 +1,94 @@
+
+#
+# Simple Machine - High level topology - Maps to machine driver.
+#
+#
+# PCM 0 <-> bufferN.0 <-> volumeN.0 <-> bufferN.1 -> SSP0
+#
+
+<include/common/tokens.conf>
+<include/platform/intel/bxt.conf>
+<include/pipelines/pipeline-volume-playback.conf>
+<include/pipelines/pipeline-volume-capture.conf>
+<include/common/connection.conf>
+<include/common/pipe_endpoint.conf>
+<include/components/dai.conf>
+<include/common/data.conf>
+<include/dais/ssp.conf>
+<include/common/manifest.conf>
+
+#
+# Pipeline definitions
+#
+
+# Pipeline ID:1 PCM ID: 0
+Object.volume-playback."1" {
+	pipeline_id	1
+	pcm_id		0
+	pcm_name	"Port0"
+	format		"s24le"
+
+	# set the mixer name for PGA.0
+	pga.0.mixer.0.name "1 Master Playback Volume"
+}
+
+# Pipeline ID:2 PCM ID: 0
+Object.volume-capture."2" {
+	pipeline_id	2
+	pcm_id		0
+	pcm_name	"Port0"
+	format		"s24le"
+
+	# set the mixer name for PGA.0
+	pga.0.mixer.0.name	"2 Master Capture Volume"
+}
+
+#
+# List of all DAIs
+#
+#SSP Index: 0, Direction: duplex
+Object.SSP."0" {
+	direction		"duplex"
+	stream_name		"NoCodec-0"
+	id 			0
+	default_hw_config	0
+	format			"s24le"
+	quirks			"lbm_mode"
+	sample_bits		24
+	Object.hw_config."0" {
+		id		0
+		mclk_freq	24000000
+		bclk_freq	4800000
+		tdm_slot_width	25
+	}
+
+	#Add DAI widgets
+	Object.dai."playback" {
+		direction	"playback"
+		widget_type	"dai_in"
+		period_source_count	2
+		period_sink_count	0
+	}
+
+	Object.dai."capture" {
+		direction	"capture"
+		widget_type	"dai_out"
+		period_source_count	0
+		period_sink_count	2
+	}
+}
+
+#
+# List of all endpoint connections
+#
+# Connect: Pipeline 1 Endpoint 0 -> SSP 0 DAI_IN
+Object.connection."0" {
+	source	"Object.volume-playback.1.endpoint.0"
+	sink	"dai.SSP.0.playback"
```

Comment:
that seems strange - why does one have `Object` and the other one doesn't?

Reply:
I explained it in the commit for the connection. Let me know if it still isnt clear. But more specfically, th reason for explicitly passing the dai widget name is that at the top-level I know the name for sure and besides the 2 DAI widgets in th SSP have the exact same index.

Reply:
I've fixed this to make it consistent with always using the Object prefix. hopefully its clearer now